server/src/main/java/org/opensearch/gateway/remote/RemoteIndexMetadataManager.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.gateway.remote;
+
+import static org.opensearch.gateway.remote.RemoteClusterStateUtils.RemoteStateTransferException;
+import static org.opensearch.gateway.remote.model.RemoteIndexMetadata.INDEX_PATH_TOKEN;
+
+import java.io.IOException;
+import java.util.Locale;
+import org.opensearch.action.LatchedActionListener;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.common.CheckedRunnable;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.gateway.remote.model.RemoteIndexMetadata;
+import org.opensearch.gateway.remote.model.RemoteIndexMetadataBlobStore;
+import org.opensearch.gateway.remote.model.RemoteReadResult;
+import org.opensearch.index.translog.transfer.BlobStoreTransferService;
+import org.opensearch.repositories.blobstore.BlobStoreRepository;
+import org.opensearch.threadpool.ThreadPool;
+
+public class RemoteIndexMetadataManager {
+
+    public static final TimeValue INDEX_METADATA_UPLOAD_TIMEOUT_DEFAULT = TimeValue.timeValueMillis(20000);
+
+    public static final Setting<TimeValue> INDEX_METADATA_UPLOAD_TIMEOUT_SETTING = Setting.timeSetting(
+        "cluster.remote_store.state.index_metadata.upload_timeout",
+        INDEX_METADATA_UPLOAD_TIMEOUT_DEFAULT,
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+
+    private final BlobStoreRepository blobStoreRepository;
+    private final ThreadPool threadPool;
+
+    private final BlobStoreTransferService blobStoreTransferService;
+    private final RemoteIndexMetadataBlobStore indexMetadataBlobStore;
+
+    private volatile TimeValue indexMetadataUploadTimeout;
+    private final String clusterName;
+
+    public RemoteIndexMetadataManager(BlobStoreRepository blobStoreRepository, ClusterSettings clusterSettings, ThreadPool threadPool, String clusterName,
+        BlobStoreTransferService blobStoreTransferService) {
+        indexMetadataBlobStore = new RemoteIndexMetadataBlobStore(blobStoreTransferService, blobStoreRepository, clusterName, threadPool);
```

Comment:
why RemoteIndexMetadataBlobStore is instantiated per entity type. BlobStore should be singleton object wired to these EntityManager classes in constructor.

Reply:
Also, RemoteIndexManager shouldn't have knowledge of blobstore otherwise we lose of benefit of defining the remote object and store interface. We wont be wire another remote store easily in RemoteIndexManager.

Reply:
Ack. Will delete RemoteIndexMetadataBlobStore and wire only the instance of RemoteBlobStore